eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -50,7 +50,9 @@ object Transactions {
 
   case class CommitTx(input: InputInfo, tx: Transaction) extends TransactionWithInputInfo
   case class HtlcSuccessTx(input: InputInfo, tx: Transaction, paymentHash: BinaryData) extends TransactionWithInputInfo
-  case class HtlcTimeoutTx(input: InputInfo, tx: Transaction) extends TransactionWithInputInfo
+  case class HtlcTimeoutTx(input: InputInfo, tx: Transaction, paymentHash: BinaryData) extends TransactionWithInputInfo
+  case class HtlcTimeoutTxLegacy(input: InputInfo, tx: Transaction) extends TransactionWithInputInfo
+
```

Comment:
Why do we need both `HtlcTimeoutTxLegacy` and `HtlcTimeoutTx` ? There should be a comment in the code here and in Helpers explaining the problem and why it is implemented this way